web_poet/requests.py

```diff
@@ -0,0 +1,132 @@
+"""This module has a full support for :mod:`asyncio` that enables developers to
+perform asynchronous additional requests inside of Page Objects.
+
+Note that the implementation to fully execute any :class:`~.Request` is not
+handled in this module. With that, the framework using **web-poet** must supply
+the implementation.
+
+You can read more about this in the :ref:`advanced-downloader-impl` documentation.
+"""
+
+import asyncio
+import logging
+from contextvars import ContextVar
+from typing import Optional, List, Callable, Union
+
+import attrs
+
+from web_poet.page_inputs import HttpResponse, HttpRequest, Headers, Body
```

Comment:
What do you think about separating types like Body and "real" page inputs like HttpResponseBody further? My worry is that user can actually add Body as a dependency, and it'd work (andi supports unions). This import reads as if Body is a page input. But it's unlikely that'd be an intention, e.g. a provider which provides bytes may break it.

How to separate them further? I can see a few options:

1) move them to a separate module. This doesn't seem to be common to move such type definitions to a separate module, but it may be ok.
2) Rename to something like BodyT or BodyType.
3) Drop it or make private, use `Union[bytes, HttpResponseBody]`
4) somethins else?

Reply:
Good catch! `Headers` and `Body` types are not being used anywhere inside the **page_inputs.py** module. They're only being used in the **requests.py** so we can move it there. In addition, we can also make them private to discourage users from importing them in their projects.

Updated this in https://github.com/scrapinghub/web-poet/pull/22/commits/9f43187c5944c0cf0ce3bc122f3c00580262dec1